samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java

```diff
@@ -278,6 +281,7 @@ private static void calculateIntStreamPartitions(JobGraph jobGraph, Config confi
     }
     for (StreamEdge edge : jobGraph.getIntermediateStreamEdges()) {
       if (edge.getPartitionCount() <= 0) {
+        log.info("Initialize the partition count for edge {} to {}.", edge.getName(), partitions);
```

Comment:
There should be another place where we set the default partition count for edges that we could not infer. Can you add logging there too?

Reply:
Are you referring to the place where we initialize the `partitions` to be `PARTITIONS_UNKNOWN` during field declaration? 
I have all other places where we set the partition count covered.

Reply:
I mean in calculateIntStreamPartitions. I see now that you already added it, thanks. Maybe distinguish this in the log message, and clarify that it is for partition count we could not infer.